src/main/kotlin/com/ampnet/identityservice/service/impl/VeriffServiceImpl.kt

```diff
@@ -0,0 +1,259 @@
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestTemplate
+import org.springframework.web.client.postForEntity
+import java.net.URI
+import java.security.MessageDigest
+import java.time.ZonedDateTime
+
+@Service
+@Suppress("TooManyFunctions", "ReturnCount")
+class VeriffServiceImpl(
+    private val veriffSessionRepository: VeriffSessionRepository,
+    private val veriffDecisionRepository: VeriffDecisionRepository,
+    private val userInfoRepository: UserInfoRepository,
+    private val applicationProperties: ApplicationProperties,
+    private val userService: UserService,
+    private val restTemplate: RestTemplate,
+    @Qualifier("camelCaseObjectMapper") private val objectMapper: ObjectMapper
+) : VeriffService {
+
+    companion object : KLogging()
+
+    /**
+     * Service returns to the user Veriff session data depending on the user state in the session.
+     * For non existing session, new Veriff session is generated.
+     * If the user started a session and still waiting for the decision (decision is missing) from Veriff,
+     * current session is still in process and returned to the user.
+     * If the decision is received from Veriff, response depends on the decision status:
+     * for `declined`, `abandoned` or `expired` new Veriff session is generated and returned with current decision,
+     * for other statuses current session is return with current decision.
+     * For more info see: <a href="https://developers.veriff.com/#session-status-diagram">Veriff diagram</a>
+     * If the session is older than 7 days, create new session.
+     *
+     * @param address String of the user who initiated Veriff session flow.
+     * @param baseUrl String of the url from which request came from.
+     * @return ServiceVerificationResponse containing Veriff session verificationUrl and state as mandatory data,
+     * `decision` is null until Veriff sends the data to webhook. Null response is returned if
+     * the new Veriff session cannot be created.
+     * @throws ResourceNotFoundException if the address is missing.
+     */
+    @Transactional
+    @Throws(ResourceNotFoundException::class)
+    override fun getVeriffSession(address: String, baseUrl: String): ServiceVerificationResponse? {
+        logger.debug { "Get Veriff session for address: $address" }
+        val session = veriffSessionRepository.findByUserAddressOrderByCreatedAtDesc(address).firstOrNull()
+            ?: return createVeriffSession(address, baseUrl)?.let { newSession ->
+                ServiceVerificationResponse(newSession.url, newSession.state)
+            }
+        logger.debug { "User has pending Veriff session" }
+
+        @Suppress("MagicNumber")
+        if (session.createdAt.isBefore(ZonedDateTime.now().minusDays(7))) {
```

Comment:
Number of days should be a constant or an application property. Using ` @Suppress("MagicNumber")` here is kind of a code smell.

Reply:
It is the duration of veriff session, it is fixed by a third party provider (Veriff). I will extract it into a constant.